src/laboratory/shelfobject/serializers.py

```diff
@@ -20,3 +22,15 @@ class TransferOutShelfObjectSerializer(serializers.Serializer):
     amount_to_transfer = serializers.FloatField()
     mark_as_discard = serializers.BooleanField()
     laboratory = serializers.PrimaryKeyRelatedField(queryset=Laboratory.objects.all())
+
+
+class ShelfObjectDeleteSerializer(serializers.Serializer):
+    shelfobj = serializers.PrimaryKeyRelatedField(queryset=ShelfObject.objects.all())
+
+    def validate_shelfobj(self, value):
+        attr = super().validate(value)
+        if attr.in_where_laboratory_id != self.context.get('laboratory').pk:
```

Comment:
Maybe you should only send the pk in the context, instead of the whole thing, it will save some resources since you only use the pk.